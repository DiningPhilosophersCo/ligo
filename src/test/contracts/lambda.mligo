type storage = unit

(* not supported yet
let main (p, s : unit * storage) = (fun x -> ()) ()
*)

let main (p, s : unit * storage) = (fun (_ : unit) -> ()) ()
