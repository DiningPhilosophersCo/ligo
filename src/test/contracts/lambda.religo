type storage = unit;

/* not supported yet
   let%entry main (p:unit) storage =
     (fun x -> ()) ()
   */

let main = ((p,s) : (unit, storage)) : unit =>
  (((useless : unit)) => ()) ();
