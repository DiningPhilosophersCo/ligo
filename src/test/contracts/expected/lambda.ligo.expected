function f (const x : unit) : unit is Unit

function main (const p : unit; const s : unit) : unit is
  f (Unit)
